nx_cugraph/convert_matrix.py

```diff
@@ -276,3 +284,95 @@ def from_scipy_sparse_array(
     if inplace:
         return create_using._become(G)
     return G
+
+
+@networkx_algorithm(version_added="25.06", fallback=True)
```

Comment:
We'll probably want to remove `fallback=True` once this PR is finished, but it's okay to use during development if it helps with testing.

`fallback=True` is primarily (and I think only) used for functions that _create_ a graph. It is a crude and somewhat magical try-except that catches any exception and turns it into NotImplementedError so that networkx dispatch machinery will "fall back" and use other backends such as default NetworkX. The idea is that if data ingestion is unable to make a CudaGraph (our data model doesn't support the same universe as networkx does), then it's better to create a NetworkX graph, which we can then hopefully accelerate for specific edge attributes.

If you want to see how magical this is, take a look at `_And_NotImplementedError` class!